src/task_system/task_system_server/Controllers/TaskSystemController.cs

```diff
@@ -26,37 +26,41 @@ public TaskSystemController(ITaskSystemRepository taskSystemRepository, IPermiss
         [HttpGet]
         public async Task<IActionResult> GetTasks([FromQuery] QueryObject query)
         {
-            if (!Request.Headers.TryGetValue("X-API-Key", out var apiKey) || !_permissionValidator.ValidatePermissions(PermissionNames.VIEW_TASKS, apiKey!))
+            if (!Request.Headers.TryGetValue("X-API-Key", out var apiKey))
                 return Unauthorized(new TaskSystemResponse<string>(ResponseMessages.UNAUTHORIZED, null));
 
+            if(!_permissionValidator.ValidatePermissions(PermissionNames.VIEW_TASKS, apiKey) && 
```

Comment:
```
(!_permissionValidator.ValidatePermissions(PermissionNames.VIEW_TASKS, apiKey)
```
apiKey may be null here, please add apiKey**!** to suppress warnings. (note, we know it's not null since handled in the tryGetValue if-statement.)